src/Rainbow/Storage/SerializationFileSystemDataStore.cs

```diff
@@ -32,10 +32,10 @@ public SerializationFileSystemDataStore(string physicalRootPath, bool useDataCac
 			// ReSharper disable once DoNotCallOverridableMethodsInConstructor
 			PhysicalRootPath = InitializeRootPath(physicalRootPath);
 
-			_useDataCache = useDataCache;
+			UseDataCache = useDataCache;
```

Comment:
Question: I don't know what your end usage is, but assuming your overridden class also receives these constructor parameters couldn't you nab them that way instead of making these protected?

ex:
```
private bool _haha;
public Foo(bool lol, string haha) : base(lol, haha) {
    _haha = haha;
}
```